weave/weave_client.py

```diff
@@ -662,8 +667,13 @@ def _save_object_basic(
         return ref
 
     def _save_nested_objects(self, obj: Any, name: Optional[str] = None) -> Any:
-        if get_ref(obj) is not None:
-            return
+        ref = get_ref(obj)
+        if ref is not None:
+            if ref.project == self.project:
```

Comment:
```suggestion
        if (ref := get_ref(obj)) is not None and ref.project == self.project:
```

Reply:
Woah do we not support 3.7? Sick @andrewtruong